src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1124,48 +1140,60 @@ private static int PathContainsSubstring(string pathToScan, string pathToLookFor
         /// <param name="pathToAdd">Collection of individual paths to add.</param>
         /// <param name="insertPosition">-1 to append to the end; 0 to insert in the beginning of the string; etc...</param>
         /// <returns>Result string.</returns>
-        private static string AddToPath(string basePath, string pathToAdd, int insertPosition)
+        private static string UpdatePath(string basePath, string pathToAdd, ref int insertPosition)
         {
             // we don't support if any of the args are null - parent function should ensure this; empty values are ok
-            Diagnostics.Assert(basePath != null, "basePath should not be null according to contract of the function");
-            Diagnostics.Assert(pathToAdd != null, "pathToAdd should not be null according to contract of the function");
+            Dbg.Assert(basePath != null, "basePath should not be null according to contract of the function");
+            Dbg.Assert(pathToAdd != null, "pathToAdd should not be null according to contract of the function");
 
-            StringBuilder result = new StringBuilder(basePath);
+            if (string.IsNullOrEmpty(pathToAdd))
+            {
+                // we don't want to append empty paths.
+                return basePath;
+            }
 
-            if (!string.IsNullOrEmpty(pathToAdd)) // we don't want to append empty paths
+            StringBuilder result = new(basePath);
+            // The 'pathToAdd' could be a 'combined path' (semicolon-separated).
+            foreach (string subPathToAdd in pathToAdd.Split(Path.PathSeparator, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries))
             {
-                foreach (string subPathToAdd in pathToAdd.Split(Path.PathSeparator, StringSplitOptions.RemoveEmptyEntries)) // in case pathToAdd is a 'combined path' (semicolon-separated)
+                if (string.IsNullOrEmpty(subPathToAdd))
                 {
-                    int position = PathContainsSubstring(result.ToString(), subPathToAdd); // searching in effective 'result' value ensures that possible duplicates in pathsToAdd are handled correctly
-                    if (position == -1) // subPathToAdd not found - add it
-                    {
-                        if (insertPosition == -1 || insertPosition > basePath.Length) // append subPathToAdd to the end
-                        {
-                            bool endsWithPathSeparator = false;
-                            if (result.Length > 0)
-                            {
-                                endsWithPathSeparator = (result[result.Length - 1] == Path.PathSeparator);
-                            }
+                    // Extreme case: empty path after trimming - skip it.
```

Comment:
It seems TrimEntries works before RemoveEmptyEntries so we shouldn't do the check and continue.